dynamodb/src/main/scala/uk/gov/nationalarchives/DADynamoDBClient.scala

```diff
@@ -31,6 +31,30 @@ class DADynamoDBClient[F[_]: Async](dynamoDBClient: DynamoDbAsyncClient):
   extension [T](completableFuture: CompletableFuture[T])
     private def liftF: F[T] = Async[F].fromCompletableFuture(Async[F].pure(completableFuture))
 
+  /** Deletes a list of items with primary keys of type T from Dynamo. Unprocessed items will be retried
+    *
+    * @param tableName
+    *   The name of the table
+    * @param primaryKeys
+    *   A list of primary keys to delete from Dynamo (list can be any length; requests will be batched into groups of
+    *   25)
+    * @param format
+    *   An implicit scanamo formatter for type T
+    * @tparam T
```

Comment:
Should we mention that T should be a case class?

Reply:
I don't really want it to be a case class only, we've had trouble with that before trying to pass traits in. So long as you have a valid `DynamoFormat[T]` you should be able to pass whatever you want in here.